packages/core/snapshot/manager.go

```diff
@@ -0,0 +1,175 @@
+package snapshot
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/iotaledger/goshimmer/packages/core/epoch"
+	"github.com/iotaledger/goshimmer/packages/core/ledger"
+	"github.com/iotaledger/goshimmer/packages/core/notarization"
+	"github.com/iotaledger/goshimmer/packages/core/tangleold"
+	"github.com/iotaledger/goshimmer/packages/node/database"
+	"github.com/iotaledger/hive.go/core/generics/event"
+	"github.com/iotaledger/hive.go/core/kvstore"
+)
+
+const (
+	prefixSolidEntryPoint byte = iota
+)
+
+// Manager is the snapshot manager.
+type Manager struct {
+	tangle          *tangleold.Tangle
+	notarizationMgr *notarization.Manager
+
+	baseStore kvstore.KVStore
+	sepsLock  sync.RWMutex
+}
+
+// NewManager creates and returns a new snapshot manager.
+func NewManager(store kvstore.KVStore, t *tangleold.Tangle, nmgr *notarization.Manager) (new *Manager) {
+	new = &Manager{
+		tangle:          t,
+		notarizationMgr: nmgr,
+	}
+
+	new.baseStore = store
+
+	new.tangle.ConfirmationOracle.Events().BlockAccepted.Attach(event.NewClosure(func(e *tangleold.BlockAcceptedEvent) {
```

Comment:
Could we attach to these from the plugin? This is usually how events are wired up.